service/history/workflow_task_handler_callbacks.go

```diff
@@ -646,7 +661,7 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 			bufferedEventShouldCreateNewTask ||
 			activityNotStartedCancelled {
 			newWorkflowTaskType = enumsspb.WORKFLOW_TASK_TYPE_NORMAL
-		} else if weContext.UpdateRegistry(ctx).HasOutgoing() {
+		} else if weContext.UpdateRegistry(ctx).HasOutgoingMessages(true) { // If there are updates which were sent on another WT (which was lost), resend them with this new WT.
```

Comment:
if current workflow task completed successfully, there should be no update left in `sent` state here?

Reply:
Yes, I fixed the comment. It there shouldn't be any updates in `stateSent` at this moment.